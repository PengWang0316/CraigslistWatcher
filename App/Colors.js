export default {
  primary: '#4f6d7a',
  primaryDark: '#24424e',
  primaryLight: '#7c9ba9'
};
