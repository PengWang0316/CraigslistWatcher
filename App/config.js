export const AMOUNT_ITEMS_IN_ONE_PAGE = 120;
