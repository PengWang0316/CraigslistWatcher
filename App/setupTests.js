// import 'raf/polyfill';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mocks from 'react-native-jest-mocks';

configure({ adapter: new Adapter() });
mocks.initAll();
// Object.defineProperty(document, 'currentScript', {
//   value: document.createElement('script'),
// });
