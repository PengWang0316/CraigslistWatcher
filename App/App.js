import React from 'react';
import { TabNavigator, TabBarBottom } from 'react-navigation';
import Ionicons from 'react-native-vector-icons/Ionicons';

import CraigslistExploreStack from './screens/CraigslistExploreStack';
import WatchingListStack from './screens/WatchingListStack';
import Colors from './Colors';

const App = TabNavigator(
  {
    CraigslistExplore: { screen: CraigslistExploreStack },
    WatchingList: { screen: WatchingListStack }
  },
  {
    navigationOptions: ({ navigation }) => ({
      tabBarIcon: ({ focused, tintColor }) => {
        const { routeName } = navigation.state;
        let iconName;
        if (routeName === 'CraigslistExplore') {
          iconName = `ios-images${focused ? '' : '-outline'}`;
        } else if (routeName === 'WatchingList') {
          iconName = `ios-eye${focused ? '' : '-outline'}`;
        }
        return <Ionicons name={iconName} size={25} color={tintColor} />;
      },
    }),
    tabBarOptions: {
      activeTintColor: 'white',
      inactiveTintColor: 'lightgray',
      style: {
        backgroundColor: Colors.primary,
      }
    },
    tabBarComponent: TabBarBottom,
    tabBarPosition: 'bottom',
    animationEnabled: true,
    swipeEnabled: true
  }
);
export default App;
