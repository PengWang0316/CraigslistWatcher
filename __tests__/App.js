import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App/App';

// jest.mock('react-navigation', () => ({ TabNavigator: jest.fn(), TabBarBottom: null }));
jest.mock('../App/screens/CraigslistExploreStack', () => 'CraigslistExploreStack');
jest.mock('../App/screens/WatchingListStack', () => 'WatchingListStack');
jest.mock('react-native-vector-icons/Ionicons', () => 'Ionicons');

// test('renders correctly', () => {
//   const reactNavigation = require('react-navigation');
//   const mockFn = jest.fn();
//   reactNavigation.TabNavigator = mockFn;
//   const App = require('../App/App');
//   expect(mockFn).toHaveBeenCalledTimes(1);
// });

test('snapshot', () => expect(renderer.create(<App />).toJSON()).toMatchSnapshot());
