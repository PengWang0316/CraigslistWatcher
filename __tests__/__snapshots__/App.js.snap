// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    automaticallyAdjustContentInsets={false}
    bounces={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    overScrollMode="never"
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
        testID={undefined}
      >
        <View
          collapsable={false}
          removeClippedSubviews={false}
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <CraigslistExploreStack
              navigation={
                Object {
                  "addListener": [Function],
                  "dispatch": [Function],
                  "getParam": [Function],
                  "goBack": [Function],
                  "isFocused": [Function],
                  "navigate": [Function],
                  "pop": [Function],
                  "popToTop": [Function],
                  "push": [Function],
                  "replace": [Function],
                  "setParams": [Function],
                  "state": Object {
                    "key": "CraigslistExplore",
                    "params": undefined,
                    "routeName": "CraigslistExplore",
                  },
                }
              }
              screenProps={undefined}
            />
          </View>
        </View>
      </View>
      <View
        style={null}
        testID={undefined}
      />
    </View>
  </RCTScrollView>
  <View
    collapsable={undefined}
    style={undefined}
  >
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#4f6d7a",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 29,
                },
                false,
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 30,
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Ionicons
                color="white"
                name="ios-images"
                size={25}
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 30,
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Ionicons
                color="lightgray"
                name="ios-images-outline"
                size={25}
              />
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            CraigslistExplore
          </Text>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 29,
                },
                false,
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 30,
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Ionicons
                color="white"
                name="ios-eye"
                size={25}
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "minWidth": 30,
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Ionicons
                color="lightgray"
                name="ios-eye-outline"
                size={25}
              />
            </View>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            collapsable={undefined}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(211, 211, 211, 1)",
                "fontSize": 10,
                "marginBottom": 1.5,
                "textAlign": "center",
              }
            }
          >
            WatchingList
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
